File: go/string-to-int/string-to-int.go
package main

import (
	"fmt"
	"strings"
)

func strToByte(s string)[]byte{
	return []byte(s)
}

func myAtoi(s string)int{
	sSlice := strings.TrimSpace(s)
	if len(sSlice) == 0{
		return 0
	}

	sign := 1
	i := 0
	res := 0
	negative := "-"
	negativeByte := strToByte(negative)
	positive := "+"
	positiveByte := strToByte(positive)
	zero := "0"
	zeroByte := strToByte(zero)
	nine := "9"
	nineByte := strToByte(nine)

	if sSlice[i] == negativeByte[0]{
        sign = -1
        i++
    }else if sSlice[i] == positiveByte[0]{
        i++
    }


	for i < len(sSlice) && sSlice[i] >= zeroByte[0] && sSlice[i] <= nineByte[0]{
		res = res * 10 + int(sSlice[i]-zeroByte[0])
		if sign * res > (1<<31-1){ return 1<<31-1
		}
		if sign * res < -(1<<31){ 
			return -(1<<31)
		}
		i++
	}

	return res * sign
}

func main(){
	s := "+1" 

	fmt.Println("answer to the first example is:",myAtoi(s))

}

File: js/string-to-int/string-to-int.js
function myAtoi(s){
    s = s.trim()
    if (s.length == 0) return 0

    let sign = 1, i =0, res = 0
    if (s[i] == "-"){
        sign = -1
        i++
    }else if (s[i] == "+"){
        i++
    }

    while (i < s.length && s[i] >= "0" && s[i] <= "9"){
        res = res * 10 + (s[i]-"0")
        if(sign * res > 2**31-1) return 2**31-1
        if(sign * res < -(2**31)) return -(2**31)
        i++
    }
    return sign * res
}

let s = "4193 with words"
console.log("answer to the first example is:",myAtoi(s))